chore: add a comment to address a potential bug that is introduced by orphaned ipconfigs on the lb

nilo19 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

/kind bug

#### What this PR does / why we need it:

There is a corner case that the VMSS VM is deleted but it is still in the lb backend pool. In this case the cloud provider keep  searching this VM by refreshing the cache, which introduces a lot of unexpected API calls. This PR adds a nil entry to the cache for the not found VM so next time when we try to search the VM it will hit the nil entry and return without calling the LIST VMSS VM API.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

andyzhangx reviewed (commented):
there is condition that if vmss instance entry is set as nil, it would never be refreshed until 15min time expire, and when the same vmss instance is recreated next time, the cloud provider cannot get this cache until 15min time expire?

feiskyer commented:
> there is condition that if vmss instance entry is set as nil, it would never be refreshed until 15min time expire, and when the same vmss instance is recreated next time, the cloud provider cannot get this cache until 15min time expire?

That's a good catch. It would introduce issues for newly created nodes and newly created Node object may be deleted because `nil` equals 'instance not found'.

nilo19 commented:
/release-note-none

nilo19 commented:
> > there is condition that if vmss instance entry is set as nil, it would never be refreshed until 15min time expire, and when the same vmss instance is recreated next time, the cloud provider cannot get this cache until 15min time expire?
> 
> That's a good catch. It would introduce issues for newly created nodes and newly created Node object may be deleted because `nil` equals 'instance not found'.

Thanks for the catch. I thought the newly created vm cannot be the same as before, but I ignore the reimage case. I have removed the change and added a comment to address the unfixable issue.

andyzhangx reviewed (approved):
/lgtm
let's brainstorm other way to fix this throttling issue. thanks.
